source/Halibut.Tests/UsageFixture.cs

```diff
@@ -37,38 +37,98 @@ public async Task OctopusCanSendMessagesToTentacle_WithEchoService(ClientAndServ
+                action.Should().Throw<HalibutClientException>();
+            }
+        }
+
+        [Test]
+        [TestCaseSource(typeof(LatestAndPreviousClientAndServiceVersionsNoWebSocketPerfectNetworkTestCases))]
+        // PollingOverWebSockets does not support (or use) ProxyDetails if provided.
+        public async Task ClientCanSendMessagesToTentacle_WithEchoService_AndPortForwarding_AndProxy(ClientAndServiceTestCase clientAndServiceTestCase)
         {
-            using (IClientAndService clientAndService = await testCase.CreateTestCaseBuilder()
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
                        .WithStandardServices()
+                       .WithPortForwarding(i => PortForwarderUtil.ForwardingToLocalPort(i).Build())
                        .WithProxy()
                        .WithHalibutLoggingLevel(LogLevel.Info)
                        .Build())
             {
                 var echo = clientAndService.CreateClient<IEchoService>();
                 echo.SayHello("Deploy package A").Should().Be("Deploy package A...");
 
-                for (var i = 0; i < 10; i++)
+                for (var i = 0; i < 5; i++)
                 {
                     echo.SayHello($"Deploy package A {i}").Should().Be($"Deploy package A {i}...");
                 }
             }
         }
 
         [Test]
-        [TestCaseSource(typeof(LatestAndPreviousClientAndServiceVersionsTestCases))]
-        // PollingOverWebSockets does not support (or use) ProxyDetails if provided. If support is added, test variations should be added
-        public async Task OctopusCanNotSendMessagesToTentacle_WithEchoService_AndABrokenProxy(ClientAndServiceTestCase testCase)
+        [TestCaseSource(typeof(LatestAndPreviousClientAndServiceVersionsNoWebSocketsTestCases))]
+        // PollingOverWebSockets does not support (or use) ProxyDetails if provided.
+        public async Task OctopusCanSendMessagesToTentacle_WithEchoService_AndAProxy(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithStandardServices()
+                       .WithProxy()
+                       .WithHalibutLoggingLevel(LogLevel.Info)
+                       .Build())
+            {
+                var echo = clientAndService.CreateClient<IEchoService>();
+                echo.SayHello("Deploy package A").Should().Be("Deploy package A...");
+
+                for (var i = 0; i < 5; i++)
+                {
+                    echo.SayHello($"Deploy package A {i}").Should().Be($"Deploy package A {i}...");
+                }
+            }
+        }
+
+        [Test]
+        [TestCaseSource(typeof(LatestAndPreviousClientAndServiceVersionsNoWebSocketPerfectNetworkTestCases))]
+        // PollingOverWebSockets does not support (or use) ProxyDetails if provided.
```

Comment:
TODO might need to come back to this one since I think the `LatestAndPreviousClientAndServiceVersionsNoWebSocketPerfectNetworkTestCases` has proxy support....